grpc-sys/build.rs

```diff
@@ -11,15 +11,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#![allow(dead_code, unused_imports)]
+
 extern crate gcc;
+extern crate cmake;
+extern crate pkg_config;
+
+use std::path::Path;
+use std::process::Command;
+
+const GRPC_VERSION: &'static str = "1.2.5";
+const ZLIB_VERSION: &'static str = "1.2.8";
+const BORINGSSL_GIT_HASH: &'static str = "78684e5b222645828ca302e56b40b9daff2b2d27";
+
+#[cfg(not(feature = "static-link"))]
+fn build_or_link_grpc(cc: &mut gcc::Config) {
+    if let Ok(lib) = pkg_config::Config::new()
+        .atleast_version(GRPC_VERSION)
+        .statik(true)
+        .probe("grpc_unsecure") {
+        for inc_path in lib.include_paths {
+            cc.include(inc_path);
+        }
+    } else {
+        panic!("can't find dynamic grpc library");
+    }
+}
+
+#[cfg(feature = "static-link")]
+fn build_or_link_grpc(cc: &mut gcc::Config) {
+    fn wget(url: &str, out: &str) -> Result<(), String> {
+        Command::new("wget")
+            .args(&["-q", "-c", "-O", out, url])
+            .status()
+            .map_err(|err| format!("wget execute failed: {}", err))
+            .and_then(|status| {
+                if status.success() {
+                    Ok(())
+                } else {
+                    Err(format!("wget exit with {}", status))
+                }
+            })
+    }
+
+    fn tar_xf(file: &str) -> Result<(), String> {
+        Command::new("tar")
+            .args(&["zxf", file])
+            .status()
+            .map_err(|err| format!("tar execute failed: {}", err))
+            .and_then(|status| {
+                if status.success() {
+                    Ok(())
+                } else {
+                    Err(format!("tar exit with {}", status))
+                }
+            })
+    }
+
+    if !Path::new(&format!("grpc-{}", GRPC_VERSION)).exists() {
+        wget(&format!("https://github.com/grpc/grpc/archive/v{}.tar.gz",
+                      GRPC_VERSION),
+             "grpc.tar.gz")
+            .and_then(|_| tar_xf("grpc.tar.gz"))
+            .unwrap();
+    }
+
+    if !Path::new(&format!("zlib-{}", ZLIB_VERSION)).exists() {
```

Comment:
Should put the directory in target directory. You can get the path by reading the environment variable `OUT_DIR`.